### PR TITLE
CLI listtransactions command

### PR DESCRIPTION
**Pull request by Ximik**

### CODE REVIEW

File: node/src/bin/space-cli.rs [hunk 1]
```diff
@@ -207,6 +207,14 @@ enum Commands {
         #[arg(long, short)]
         fee_rate: Option<u64>,
     },
+    /// List last transactions
+    #[command(name = "listtransactions")]
+    ListTransactions {
+        #[arg(default_value = "1")]
```
Comment (buffrr) [inline node/src/bin/space-cli.rs:213]: lists only 1 tx by default?

File: node/src/wallets.rs [hunk 2]
```diff
@@ -51,6 +50,12 @@ pub struct TxResponse {
     pub error: Option<BTreeMap<String, String>>,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TxInfo {
+    pub txid: Txid,
```
Comment (buffrr) [inline node/src/wallets.rs:55]: is the txid and confirmed enough information for the wallet? shouldn't it display at least sent or received amount in this tx?

File: protocol/src/lib.rs [hunk 3]
```diff
@@ -138,14 +135,14 @@ pub enum RevokeReason {
...
-    BidPSBT(BidPsbtReason),
+    BidPsbt(BidPsbtReason),
 }
 
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
 #[cfg_attr(
     feature = "serde",
     derive(Serialize, Deserialize),
-    serde(rename_all = "snake_case", tag = "bid_psbt_reason")
+    serde(rename_all = "snake_case")
```
Comment (buffrr) [inline protocol/src/lib.rs:145]: do you have an example of this looks like as JSON now for revoke and reject?

**Review by buffrr — CHANGES_REQUESTED** [review]
Thanks alex i added a few comments

File: protocol/src/lib.rs (hunk 3 above)
Comment (Ximik) [reply protocol/src/lib.rs:145]: Here are the responses examples
```
{"type":"reject","name":"@artfuldev","reason":"already_exists"}
{"type":"reject","name":"@tap4","reason":{"bid_psbt":"output_spent"}}
```

Comment (buffrr) [reply protocol/src/lib.rs:145]: hm so the reason is now either an object or a string previously for revoke it was like this 

```json
{ "revoke_reason": "bid_psbt", "bid_psbt_reason": "output_spent" } 
```

could we make the reject look the same in a simple way e.g.

```json
{"type":"reject","name":"@artfuldev","reason":"already_exists"}
{"type":"reject","name":"@tap4","reason": "bid_psbt", "bid_psbt_reason":"output_spent"}
```

File: node/src/bin/space-cli.rs (hunk 1 above)
Comment (Ximik) [reply node/src/bin/space-cli.rs]: Set to 10

File: node/src/wallets.rs (hunk 2 above)
Comment (Ximik) [reply node/src/wallets.rs:55]: I've added some more data.

File: protocol/src/lib.rs (hunk 3 above)
Comment (Ximik) [reply protocol/src/lib.rs]: Fixed in #30

**Review by buffrr — APPROVED** [review]
Thanks LGTM

**Pull request merged**